src/main/java/game/map/MapValidator.java

```diff
@@ -0,0 +1,176 @@
+	 */
+	public List<Boolean> dfs(Integer p_startingCountryID, Continent p_continent) {
+		List<Boolean> l_isVisited = new ArrayList<Boolean>();
+		for (int i=0; i<p_continent.getCountryIdList().size(); i++)
+			l_isVisited.add(Boolean.FALSE);
+		return dfsStep(p_startingCountryID, p_continent, l_isVisited);
+	}
+
+	/**
+	 * Recursive function for the DFS. The algorithm flags the current country as visited and the function is
+	 * called recursively for each neighbor that hasn't been seen yet.
+	 * @param p_currentCountryID Id of the country being visited.
+	 * @param p_continent Continent where the DFS is applied
+	 * @param p_countryHasBeenSeen List representing if each country is seen or not
+	 * @return The list where we can see if a country has been visited or not
+	 */
+	private List<Boolean> dfsStep(Integer p_currentCountryID, Continent p_continent, List<Boolean> p_countryHasBeenSeen) {
+		int l_indexCurrentCountry = p_continent.getCountryIdList().indexOf(p_currentCountryID);
+		p_countryHasBeenSeen.set(l_indexCurrentCountry, Boolean.TRUE);
+		// We call dfsStep on each neighbor (in the continent) that has not been seen yet
+		for (Integer l_neighbourID: getCountryWithId(d_mapToValidate, p_currentCountryID).getNeighborIdList()) {
+			Country l_neighbour = getCountryWithId(d_mapToValidate, l_neighbourID);
+			if (!l_neighbour.getContinent().equals(p_continent))
+				continue;
+			int l_indexNeighbour = p_continent.getCountryIdList().indexOf(l_neighbourID);
+			if (!p_countryHasBeenSeen.get(l_indexNeighbour)) {
+				dfsStep(l_neighbourID, p_continent, p_countryHasBeenSeen);
+			}
+		}
+		return p_countryHasBeenSeen;
+	}
+	
+	/**
+	 * Checks whether or not the continent is a connected subgraph, using the same process as in the mapIsConnected() method
+	 * @param p_continent The continent we want to evaluate.
+	 * @return true if the continent is a connected subgraph, false otherwise.
+	 */
+	public boolean continentIsConnected(Continent p_continent) {
+		for(Integer l_countryId: p_continent.getCountryIdList()) {
+			List<Boolean> l_dfsResult = dfs(l_countryId, p_continent);
+			for (Boolean l_isVisited: l_dfsResult) {
+				if (!l_isVisited)
+					return false;
+			}
+		}
+		return true;
+	}
+	
+	/**
+	 * Checks if the map and each continent fulfill the conditions about being connected.
+	 * @return true if everything is well connected, false otherwise.
+	 */
+	public boolean mapAndContinentsConnected() {
+		for (Continent l_continent: d_mapToValidate.getContinents()) {
+			if (!continentIsConnected(l_continent))
+				return false;
+		}
+		return mapIsConnected();
+	}
+	
+	/**
+	 * Verifies if the map is valid. </br>
+	 * We consider that a map is correct when it represents a connected graph, each continent
+	 * also represents a connected subgraph, there are no empty continents and each country has to
+	 * have a unique name and id.
+	 * @return true if the map is valid, false otherwise.
+	 */
+	public boolean isMapValid() {
```

Comment:
I think we can pass the map reference here, instead of having `private Map d_mapToValidate`. As user enters validate command while editing the map (https://github.com/shanmukhatummala/game-engine/blob/2c344b8630ed73740b2e8e7d5691e2d5e6fa5452/src/main/java/game/map/MapEditor.java#L27).

Reply:
I think we don't really need to, because the MapValidator should have the correct reference, which means it sould also receive the modifications. But your idea is probably better because then we just need one object to validate any kind of map